packages/cardpay-reward-programs/tests/rules/test_min_other_merchants_paid.py

```diff
@@ -50,60 +46,48 @@ def rule(request):
 
 class TestMinOtherMerchantsPaidSingle:
     @pytest.mark.parametrize(
-        "rule,ans",
+        "rule,summary",
         zip(
             itertools.product(payment_cycle_length_ls, min_other_merchants_ls),
-            ans_ls,
+            summaries,
         ),
         indirect=["rule"],
     )
-    def test_run(self, rule, ans, indexed_data):
-        df_hash, summary = ans
+    def test_run(self, rule, summary, indexed_data):
         start_block = 24000000
         end_block = 26000000
-        df = rule.run(start_block, end_block)
-        payment_list = rule.df_to_payment_list(df)
-        h = hashlib.sha256(pd.util.hash_pandas_object(df, index=True).values).hexdigest()
+        rule.payment_cycle_length = end_block - start_block
+        payment_list = rule.run(end_block, "0x0")
         computed_summary = rule.get_summary(payment_list)
         assert computed_summary["total_reward"][0] == summary["total_reward"]
         assert computed_summary["unique_payee"][0] == summary["unique_payee"]
-        assert h == df_hash
 
 
-multiple_df_hashes = [
-    "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
-    "f45c509908a1eba5eab02d30112b3d5153a9939df3d0e58cb4a11fadb11272c8",
-    "50bb631ce5832059f9d55de834af8aef712970adbcbe64696d261771967975c8",
-]
+
 multiple_summaries = [
     {"total_reward": 0, "unique_payee": 0},
     {"total_reward": 10, "unique_payee": 1},
-    {"total_reward": 20, "unique_payee": 2},
+    {"total_reward": 30, "unique_payee": 2},
```

Comment:
I want to verify these, but I am fairly sure these were incorrect before - because the aggregation of multiple payment cycles was then having further logic applied it was combining things across payment cycles.

Reply:
The previous test was right, if you consider `aggregate` as a way to combine the rule of two or more ranges. 

for example, 

`aggregate` of: 
- `start_block_1 < x <= start_block_2`
- `start_block_2< x <= start_block_3`

will yield 
`aggregate` of 
-`start_block_1` < x <= `start_block_2`

Perhaps unnecessarily complex. Let's forget about this logic.